GraphBase/src/graph/Vertex.java

```diff
@@ -0,0 +1,43 @@
+package graph;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.NoSuchElementException;
+
+public class Vertex {
+
+  private int vertexId;
+  private Map<Integer, Edge> adjEdges;
+
+  public Vertex(int newVertexId) {
+    this.vertexId = newVertexId;
+    this.adjEdges = new HashMap<Integer, Edge>();
+  }
+
+  public int getId() {
+    return vertexId;
+  }
+
+  public Map getAdjEdges() {
+    return adjEdges;
+  }
+
+  public void addAdjEdge(Edge newEdge) {
+    if (newEdge.getFromId() != vertexId) {
+      throw new VerifyError("The edge is not from this vertex");
```

Comment:
I am very curious. For edge (A, B) in an undirected graph, would this edge be the same, or not the same in A's adjacency list and B's adjacency list. In your case, you should definitely define two Edge objects for A and B's adjacency list. You do need to modify this, but try to think it hard which design may be better.

Reply:
Should I use different Edge class for undirected  graph and directed graph.

Reply:
They should be the same.